### PR TITLE
Expose LoadLatestOptions, LoadOptionsFromFile and GetLatestOptionsFileName APIs in RocksJava

### PR DESCRIPTION
**Pull request by sagar0**
JNI wrappers for LoadLatestOptions, LoadOptionsFromFile and GetLatestOptionsFileName APIs.

Test Plan:
Added new junit tests.

### CODE REVIEW

**Comment by siying** [comment]
Is there any .md file to update for the adding the API?

**Comment by sagar0** [comment]
There is [HISTORY-JAVA.md](https://github.com/facebook/rocksdb/blob/master/java/HISTORY-JAVA.md), but it hasn't been updated in over 2 years.

File: java/rocksjni/options_util.cc [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  // Column family descriptor
+  for (rocksdb::ColumnFamilyDescriptor& cfd : cf_descs) {
+    // Construct a ColumnFamilyDescriptor java object
+    jobject jcfd = rocksdb::ColumnFamilyDescriptorJni::construct(env, &cfd);
+    if (env->ExceptionCheck()) {
+      // exception occurred constructing object
+      if (jcfd != nullptr) {
+        env->DeleteLocalRef(jcfd);
+      }
+      return;
```
Comment (siying) [inline java/rocksjni/options_util.cc]: Do you want to clear the objects already put in cf_descs?

File: java/rocksjni/options_util.cc [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    // Add the object to java list.
+    jboolean rs = env->CallBooleanMethod(jcfds, add_mid, jcfd);
+    if (env->ExceptionCheck() || rs == JNI_FALSE) {
+      // exception occurred calling method, or could not add
+      if (jcfd != nullptr) {
+        env->DeleteLocalRef(jcfd);
+      }
+      return;
```
Comment (siying) [inline java/rocksjni/options_util.cc]: Same here.

**Pull request closed without merging**